### DISCUSSION TITLE
[18 gennaio 2023] Esercizio 1

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/psd2223#227] [Esami]
Progettare un circuito sequenziale con un ingresso $\Large x$ due uscite $\Large z1$ e $\Large z0$.

## Fila A

L’uscita $\Large z1$ deve essere uguale a $\Large 1$ se gli ultimi tre bit di ingresso contengono almeno due $\Large 0$, mentre $\Large z0$ deve essere $\Large 1$ se gli ultimi $\Large 3$ bit sono uguali. Non si considerino le sovrapposizioni.

Usare una $\Large ROM$ per la parte combinatoria e un $\Large FF$ di tipo $\Large JK$ per il bit più significativo. Disegnare il circuito ottenuto.

### Esempio

$$
    \Large x\ \ \ 10100000111
$$

$$
    \Large z1\ 00010010000
$$

$$
    \Large z0\ 00000010001
$$

## Fila B

L’uscita $\Large z1$ deve essere uguale a $\Large 1$ se gli ultimi tre bit di ingresso sono tutti uguali a $\Large 0$, mentre $\Large z0$ deve essere $\Large 1$ se negli ultimi $\Large 3$ bit ci sono almeno due $\Large 1$. Non si considerino le sovrapposizioni.

### Esempio

$$
    \Large x\ \ \ 000001011100
$$

$$
    \Large z1\ 001000000000
$$

$$
    \Large z0\ 000000010010
$$

> [!NOTE]
> L'uso di un $\Large FF$ di tipo $\Large JK$ per il bit più significativo è richiesto solo nelle tracce delle 2 file del canale M-Z

<details closed>
    <summary> Soluzione A-L Fila A </summary>

<div align="center">

![image](https://github.com/user-attachments/assets/c66d51b5-ef3d-44ce-866e-645a30b0c48b)
![image](https://github.com/user-attachments/assets/72c78380-8d55-4ca3-a35b-598787844861)

</div>
</details>

**Comment by Kisuke-cmd** [comment]
perche al terzo bit partendo da sinistra di z0 della fila B non ce' 1? gli ultimi 3 bit di x sono 0

**Reply by FeddyLix17** [reply]
ho corretto il testo dell'esercizio
